nemo/collections/asr/data/data_simulation.py

```diff
@@ -0,0 +1,1426 @@
+        ):
+            raise Exception("Incorrect window type provided")
+
+    def _get_speaker_ids(self) -> List[str]:
+        """
+        Randomly select speaker IDs from the loaded manifest file.
+
+        Returns:
+            speaker_ids (list): Speaker IDs
+        """
+        speaker_ids = []
+        s = 0
+        while s < self._params.data_simulator.session_config.num_speakers:
+            file = self._manifest[np.random.randint(0, len(self._manifest) - 1)]
+            speaker_id = file['speaker_id']
+            if speaker_id not in speaker_ids:  # ensure speaker ids are not duplicated
+                speaker_ids.append(speaker_id)
+                s += 1
+        return speaker_ids
+
+    def _get_speaker_samples(self, speaker_ids: List[str]) -> Dict[str, list]:
+        """
+        Get a list of the samples for each of the specified speakers.
+
+        Args:
+            speaker_ids (list): LibriSpeech speaker IDs for each speaker in the current session.
+        Returns:
+            speaker_lists (dict): Dictionary of manifest lines per speaker
+        """
+        speaker_lists = {}
+        for i in range(self._params.data_simulator.session_config.num_speakers):
+            speaker_lists[str(speaker_ids[i])] = []
+        # loop over manifest and add files corresponding to each speaker to each sublist
+        for file in self._manifest:
+            new_speaker_id = file['speaker_id']
+            if new_speaker_id in speaker_ids:
+                speaker_lists[str(new_speaker_id)].append(file)
+        return speaker_lists
+
+    def _load_speaker_sample(self, speaker_lists: List[dict], speaker_ids: List[str], speaker_turn: int) -> str:
+        """
+        Load a sample for the selected speaker ID.
+
+        Args:
+            speaker_lists (list): List of samples for each speaker in the session.
+            speaker_ids (list): LibriSpeech speaker IDs for each speaker in the current session.
+            speaker_turn (int): Current speaker turn.
+        Returns:
+            file_path (str): Path to the desired audio file
+        """
+        speaker_id = speaker_ids[speaker_turn]
+        file_id = np.random.randint(0, len(speaker_lists[str(speaker_id)]) - 1)
+        file_path = speaker_lists[str(speaker_id)][file_id]
+        return file_path
+
+    def _get_speaker_dominance(self) -> List[float]:
+        """
+        Get the dominance value for each speaker, accounting for the dominance variance and
+        the minimum per-speaker dominance.
+
+        Returns:
+            dominance (list): Per-speaker dominance
+        """
+        dominance_mean = 1.0 / self._params.data_simulator.session_config.num_speakers
+        dominance = np.random.normal(
+            loc=dominance_mean,
+            scale=self._params.data_simulator.session_params.dominance_var,
+            size=self._params.data_simulator.session_config.num_speakers,
+        )
+        for i in range(len(dominance)):
```

Comment:
this could be simplified by `dominance = [ max(x, 0) for x in dominance ]`
Since there are a lot of these, just create a function says:
```
def clamp_min_list(target_list, min_val):
    return  [ max(x, min_val) for x in target_list ]
```

then,
`clamp_min_list(dominance, 0)`